modAionBase/src/org/aion/base/db/IRepositoryCache.java

```diff
@@ -78,5 +79,5 @@
      * @param key the key at which the date will be stored
      * @param value the data to be stored
      */
-    void addStorageRow(Address address, DW key, DW value);
+    void addStorageRow(Address address, ByteArrayWrapper key, ByteArrayWrapper value);
```

Comment:
I noticed that as a result of this change, a lot of calls to it IRepositoryCache.addStorageRow has changed from:

`IRepositoryCache.addStorageRow(someAddress, new DataWord(someKey1), new DataWord(someKey2)`

to

`IRepositoryCache.addStorageRow(someAddress, new DataWord(someKey1).toWrapper(), new DataWord(someKey2).toWrapper()`

This happens all over the code and tests, i.e. StandaloneBlockchain.java:313-314. 

Would it make more sense to avoid that "add toWrapper() to everything" change by doing something like this:

- create an interface called ByteArrayWrappable with a method toWrapper()
- make IDataWord implement ByteArrayWrappable (and any other suitable class)
- either: (1) replace IRepositoryCache.addStorageRow ByteArrayWrapper arguments with ByteArrayWrappable and in each implementation, call ByteArrayWrappable.toWrapper() to get the ByteArrayWrapper
- or: (2) add a default method into IRepositoryCache: addStorageRow(Address, ByteArrayWrappable, ByteArrayWrappable) and all it does is call toWrapper() and pass it on to addStorageRow(Address address, ByteArrayWrapper key, ByteArrayWrapper value)

Reply:
We could make this comment an issue that can be verified at the end, once all stages of the PR are added.